src/HookFactory.php

```diff
@@ -62,6 +66,8 @@ function %%function_name%% {
     COMPREPLY=(`compgen -W "$RESULT" -- $cur`);
 
     __ltrim_colon_completions "$cur";
+
+    export MAILCHECK=mail_check_backup
```

Comment:
Also, you don't need to use `export` here or on line 50: `export` works by reference rather than value (plus you don't need to export to affect environment variables in the same shell environment - only for child processes).